packages/amplify-cli/src/version-gating/index.ts

```diff
@@ -0,0 +1,202 @@
+    return true;
+  }
+
+  // At this point the command is blocking so version check must be performed against current CLI version
+  // and the minimum CLI version vs. the deployed version from the root stack metadata (if any)
+
+  const meta = stateManager.getMeta(undefined, {
+    throwIfNotExist: false,
+  });
+
+  const rootStackName = _.get(meta, ['providers', 'awscloudformation', 'StackName']);
+
+  if (rootStackName === undefined) {
+    // No meta or root stack name is not present, which means that version gating cannot be enforced on an uninitialized project
+    // so assume version check passed as the rest of the CLI will validate the command in question for execution.
+    return true;
+  }
+
+  const cloudFormation: { cfn: CloudFormation } = await context.amplify.invokePluginMethod(
+    context,
+    'awscloudformation',
+    undefined,
+    'getCloudFormationSdk',
+    [context],
+  );
+
+  const cfnClient: CloudFormation = cloudFormation.cfn as CloudFormation;
+
+  const templateSummary = await cfnClient
+    .getTemplateSummary({
+      StackName: rootStackName,
+    })
+    .promise();
+
+  const metadataValue = _.get(templateSummary, ['Metadata']) || '{}';
+  const metadata = JSONUtilities.parse(metadataValue);
+  const versionGatingMetadata: VersionGatingMetadata = _.get(metadata, ['AmplifyCLI']);
+
+  // If no metadata or version info in metadata was found in the root stack template, then
+  // assume version check passed as current CLI is newer than the stack was deployed with.
+  if (
+    versionGatingMetadata === undefined ||
+    versionGatingMetadata.MinimumCompatibleCLIVersion === undefined ||
+    versionGatingMetadata.DeployedByCLIVersion === undefined
+  ) {
+    return true;
+  }
+
+  // These are always valid version numbers as CLI creating them.
+  const stackMinimumCompatibleCLIVersion = semver.coerce(versionGatingMetadata.MinimumCompatibleCLIVersion)!;
+
+  // Pick the greater minimum version
+  const minimumCompatibleCLIVersion = semver.gt(stackMinimumCompatibleCLIVersion, context.versionInfo.minimumCompatibleCLIVersion)
+    ? stackMinimumCompatibleCLIVersion
+    : semver.coerce(context.versionInfo.minimumCompatibleCLIVersion)!;
+
+  // If current version is greater than
+  if (semver.gte(context.versionInfo.currentCLIVersion, minimumCompatibleCLIVersion)) {
+    return true;
+  }
+
+  const deployedCLIVersion = semver.coerce(versionGatingMetadata.DeployedByCLIVersion)!;
+
+  context.print.warning(
+    `This project was previously deployed with Amplify CLI version ${deployedCLIVersion}. The currently running Amplify CLI version is ${context.versionInfo.currentCLIVersion}.`,
+  );
+
+  context.print.info('');
+  context.print.info(`Some features in this project require Amplify CLI version >=${minimumCompatibleCLIVersion} to function correctly.`);
+  context.print.info('Upgrade to the latest version of Amplify CLI, run: "amplify upgrade" or "npm install -g @aws-amplify/cli"');
```

Comment:
nit: can use `printer.blankLine()` and `printer.info()`

Reply:
Yeah the original PR was pre prompts package, updated.